nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/main/java/org/apache/nifi/parameter/aws/AwsSecretsManagerParameterProvider.java

```diff
@@ -0,0 +1,305 @@
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(ConfigVerificationResult.Outcome.FAILED)
+                    .verificationStepName("Fetch Parameters")
+                    .explanation("Failed to fetch parameters: " + e.getMessage())
+                    .build());
+        }
+        return results;
+    }
+
+    private List<ParameterGroup> fetchSecret(final AWSSecretsManager secretsManager, final ConfigurationContext context, final String secretName) {
+        final List<ParameterGroup> groups = new ArrayList<>();
+        final Pattern secretNamePattern = context.getProperty(SECRET_NAME_REGEX).isSet()
+                ? Pattern.compile(context.getProperty(SECRET_NAME_REGEX).getValue()) : null;
+
+        final List<Parameter> parameters = new ArrayList<>();
+
+        if (!secretNamePattern.matcher(secretName).matches()) {
+            logger.debug("Secret [{}] does not match the secret name regex {}", secretName, secretNamePattern);
+            return groups;
+        }
+
+        final GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest().withSecretId(secretName);
+        try {
+            final GetSecretValueResult getSecretValueResult = secretsManager.getSecretValue(getSecretValueRequest);
+
+            if (getSecretValueResult.getSecretString() == null) {
+                logger.debug("Secret [{}] is not configured", secretName);
+                return groups;
+            }
+
+            final ObjectNode secretObject = parseSecret(getSecretValueResult.getSecretString());
+            if (secretObject == null) {
+                logger.debug("Secret [{}] is not in the expected JSON key/value format", secretName);
+                return groups;
+            }
+
+            for (final Iterator<Map.Entry<String, JsonNode>> it = secretObject.fields(); it.hasNext(); ) {
+                final Map.Entry<String, JsonNode> field = it.next();
+                final String parameterName = field.getKey();
+                final String parameterValue = field.getValue().textValue();
+                if (parameterValue == null) {
+                    logger.debug("Secret [{}] key [{}] has no value", secretName, parameterValue);
+                    continue;
+                }
+
+                parameters.add(createParameter(parameterName, parameterValue));
+            }
+
+            groups.add(new ParameterGroup(secretName, parameters));
+
+            return groups;
+        } catch (final ResourceNotFoundException e) {
+            logger.error("Secret [{}] not found", secretName);
+            throw new IllegalStateException(String.format("Secret %s not found", secretName), e);
+        } catch (final AWSSecretsManagerException e) {
+            logger.error("Error retrieving secret [{}]", secretName);
+            throw new IllegalStateException("Error retrieving secret " + secretName, e);
+        }
```

Comment:
Recommend moving the `logger.error()` calls since the same information should be provided through the exceptions thrown.